### PR TITLE
added support for esbuild loader option

### PR DESCRIPTION
**Pull request by beqaMeqvabishvili**
Adds support for a new `esbuild.loader` option. 

Currently, loader cannot be configured and is guessed by the file extension. One possible use case is allowing users to use jsx loader for `.js` files.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -75,6 +75,7 @@ export interface UnpluginOptions {
     onResolveFilter?: RegExp
     onLoadFilter?: RegExp
     setup?: EsbuildPlugin['setup']
+    loader?: Loader
```
Comment (sxzz) [inline src/types.ts:78]: Maybe `loader` should be a function that is processed by userland, instead of a constant value.

Comment (sxzz) [reply src/types.ts:78]:
```suggestion
    loader?: () => Loader
```

Comment (sxzz) [reply src/types.ts:78]:
```suggestion
    loader?: () => Loader
```

Comment (beqaMeqvabishvili) [reply src/types.ts:78]: That makes a lot of sense. Would you like to make it `Loader | ((code:string, id: string) => Loader)` for convenience or should I stick with your suggestion?

Comment (sxzz) [reply src/types.ts:78]: It would be also great

**Comment by beqaMeqvabishvili** [comment]
I'm not sure which one's more desired:
a. Passing the result of `fs.readFile` to the loader functions `code` argument or
b. Passing the result of `plugin.transform` which we already have by the time loader is returned.
In this version I decided to go with the second option.

**Comment by beqaMeqvabishvili** [comment]
Hey @sxzz, any updates/instructions on this? 

If you are undecided on my previous comment, we could just merge the first version of  `loader?: Loader` (constant value, no function) and wait for the feedback. That way a chunk of the use case can be addressed instantly. And if a function is ever implemented it won't be a breaking change

**Pull request merged**